docs/ProblemStatementAndGoals/ProblemStatement.tex

```diff
@@ -47,31 +47,107 @@ \subsection{Inputs and Outputs}
+
+         \item \textbf{Tertiary Stakeholders: The Wider Reinforcement Learning (RL) Community}
+
+         The broader reinforcement learning community is a tertiary stakeholder in this project. Researchers and developers interested in the intersection of evolutionary algorithms and machine learning may benefit from the enhanced TPG framework. The improvements made through this project, including standard software engineering practices and expanded testing environments, will increase TPG’s accessibility and usability. Additionally, the publication of project results in international conferences could raise awareness of TPG’s capabilities, potentially encouraging wider adoption and collaboration within the RL community.
+
+
+\end{enumerate}
+
+
+
+
 \subsection{Environment}
 
-\wss{Hardware and software environment}
+The capstone project will primarily be based on the existing TPG framework environment, which will consist of the existing C++ codebase which will be refactored and modified to comply with standard software engineering principles, such as continuous integration, automated testing, and improved documentation. In addition, a key objective of the capstone is to extend the TPG framework beyond its current application in Atari video games by integrating it into \textbf{MuJoCo (Multi-Joint dynamics with Contact)}, a widely used physics engine developed by\textbf{ Google DeepMind}. MuJoCo provides highly accurate physical simulations, particularly suited for robotics. By interfacing TPG with MuJoCo, the project aims to evaluate the performance of the framework in controlling complex robotic systems, such as quadrupeds. This integration will allow for more realistic testing environments, expanding the boundaries of TPG's current capabilities. 
 
 \section{Goals}
 
+\begin{enumerate}
+    \item \textbf{TPG Framework must adhere to a modern software engineering Dev Ops pipeline}
+    
+    Set up fully automated CI pipeline to ensure all code changes in the TPG framework are continuously tested, verified and follow proper guidelines
+
+
+\textbf{Measurable Outcomes: 
+} To accomplish this goal, there should be a complete integration of a CI tool such as Github Actions or Gitlab CI. The CI pipeline should be invoked when changes are made to the framework via new PRs and should not result in blocking errors (either accept or reject changes after CI pipeline analyzes changes). 
+
+\textbf{Reasoning: } Establishing a CI pipeline will streamline the development process, reduce human error, and improve code reliability. Automating testing ensures that the framework is stable with each update, promoting scalability and maintainability
+
+\item \textbf{ The TPG Framework should be well tested}
```

Comment:
Maybe I'd reword this to "standardize software development workflows". The scope of this is bigger as unit testing falls under it and we're giving them a framework to allow them to do their dev